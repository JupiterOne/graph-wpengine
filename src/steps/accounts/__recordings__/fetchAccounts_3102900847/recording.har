{
  "log": {
    "_recordingName": "fetchAccounts",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "4af80a67b8b7715f75fc69566d0c1fb9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 330,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://api.wpengineapi.com/v1/user"
        },
        "response": {
          "bodySize": 119,
          "content": {
            "mimeType": "application/json",
            "size": 119,
            "text": "{\"id\":\"3937869a-09a0-4198-a1c7-d70c451c24fc\",\"email\":\"stefan@creativice.com\",\"first_name\":\"Stefan\",\"last_name\":\"Virag\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "119"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:42 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "0e1d179c-e95f-4fd1-bb5c-f2f5a2ba2bc5"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n0pFhKIAMF53g="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2ecea-4df20f310d0da6a825d1b252"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 a0549c0c4be19f37804ee07fca51b10d.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "PEofLMwLC-krC5A1NIls-yxf1KfnUzSnwId0PjhLanbTNzmp2q9-sg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.194Z",
        "time": 669,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 669
        }
      },
      {
        "_id": "b5a41916ddb8fd6f9a367486771ee186",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/accounts?limit=10&offset=0"
        },
        "response": {
          "bodySize": 112,
          "content": {
            "mimeType": "application/json",
            "size": 112,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\",\"name\":\"j1inc\"}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "112"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:28:43 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "3578bc27-3838-4460-a26c-9ec3b6bcffab"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_n00HBeoAMFU9A="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2eceb-0fd0c4e4414a5daa040a0110"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 f6773e0e5d339061619a1aeb9d029153.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "podix2UvqsAW69pR9ut6AkR8v35SP3Ya7fWL5NdbGHDivqdXh5w2gg=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:28:42.913Z",
        "time": 1067,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1067
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
