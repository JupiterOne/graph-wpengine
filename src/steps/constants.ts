import {
  RelationshipClass,
  StepEntityMetadata,
  StepRelationshipMetadata,
} from '@jupiterone/integration-sdk-core';

export const USER_ENTITY_KEY = 'entity:user';

export enum IntegrationSteps {
  USER = 'fetch-user',
  ACCOUNTS = 'fetch-accounts',
  SITES = 'fetch-sites',
  INSTALLS = 'fetch-installs',
  DOMAINS = 'fetch-domains',
}

export const Entities: Record<
  'USER' | 'ACCOUNT' | 'SITE' | 'INSTALL' | 'DOMAIN',
  StepEntityMetadata
> = {
  USER: {
    resourceName: 'User',
    _type: 'wp_engine_user',
    _class: 'User',
  },
  ACCOUNT: {
    resourceName: 'Account',
    _type: 'wp_engine_account',
    _class: 'Account',
  },
  SITE: {
    resourceName: 'Site',
    _type: 'wp_engine_site',
    _class: 'Host',
  },
  INSTALL: {
    resourceName: 'Install',
    _type: 'wp_engine_install',
    _class: 'Application',
  },
  DOMAIN: {
    resourceName: 'Domain',
    _type: 'wp_engine_domain',
    _class: 'Domain',
  },
};

export const Relationships: Record<
  | 'USER_HAS_ACCOUNT'
  | 'ACCOUNT_HAS_SITE'
  | 'SITE_HAS_INSTALL'
  | 'ACCOUNT_HAS_INSTALL'
  | 'INSTALL_HAS_DOMAIN',
  StepRelationshipMetadata
> = {
  USER_HAS_ACCOUNT: {
    _type: 'wp_engine_user_has_account',
    _class: RelationshipClass.HAS,
    sourceType: Entities.USER._type,
    targetType: Entities.ACCOUNT._type,
  },
  ACCOUNT_HAS_SITE: {
    _type: 'wp_engine_account_has_site',
    _class: RelationshipClass.HAS,
    sourceType: Entities.ACCOUNT._type,
    targetType: Entities.SITE._type,
  },
  ACCOUNT_HAS_INSTALL: {
    _type: 'wp_engine_account_has_install',
    _class: RelationshipClass.HAS,
    sourceType: Entities.ACCOUNT._type,
    targetType: Entities.INSTALL._type,
  },
  SITE_HAS_INSTALL: {
    _type: 'wp_engine_site_has_install',
    _class: RelationshipClass.HAS,
    sourceType: Entities.SITE._type,
    targetType: Entities.INSTALL._type,
  },
  INSTALL_HAS_DOMAIN: {
    _type: 'wp_engine_install_has_domain',
    _class: RelationshipClass.HAS,
    sourceType: Entities.INSTALL._type,
    targetType: Entities.DOMAIN._type,
  },
};
