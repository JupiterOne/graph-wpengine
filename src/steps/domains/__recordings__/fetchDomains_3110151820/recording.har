{
  "log": {
    "_recordingName": "fetchDomains",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "7c0782aad0b9678323207b7733ae2209",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 352,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1/installs?limit=10&offset=0"
        },
        "response": {
          "bodySize": 398,
          "content": {
            "mimeType": "application/json",
            "size": 398,
            "text": "{\"previous\":null,\"next\":null,\"count\":1,\"results\":[{\"id\":\"30c47931-a91d-468a-a5dc-b9711f06c60b\",\"name\":\"jupiterone\",\"cname\":\"jupiterone.wpengine.com\",\"stable_ips\":null,\"status\":\"active\",\"account\":{\"id\":\"77815c5d-e25c-41a3-ba2d-29f8ceb82ac1\"},\"php_version\":\"7.3\",\"site\":{\"id\":\"4d91f2dd-9743-421e-9bcf-20dcf0a195b2\"},\"environment\":\"production\",\"primary_domain\":\"jupiterone.com\",\"is_multisite\":false}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "398"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:52:23 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "b7b1983b-d336-47c9-9d42-585f688af94a"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_rSmEgpoAMFTvg="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2f276-0778f73755cc5e831f39c56a"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 c218d36bf9cc1c36cb60fc8924dc8cc1.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "Nt00erYZkmWelAHTLRemetEmc3AU8_m76RQNpHdIXxT3_WbERmtQ9Q=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:52:22.841Z",
        "time": 750,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 750
        }
      },
      {
        "_id": "80fb0b7911edbf28dc48cf90a34041f6",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "api.wpengineapi.com"
            }
          ],
          "headersSize": 398,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "10"
            },
            {
              "name": "offset",
              "value": "0"
            }
          ],
          "url": "https://api.wpengineapi.com/v1//installs/30c47931-a91d-468a-a5dc-b9711f06c60b/domains?limit=10&offset=0"
        },
        "response": {
          "bodySize": 804,
          "content": {
            "mimeType": "application/json",
            "size": 804,
            "text": "{\"previous\":null,\"next\":null,\"count\":5,\"results\":[{\"name\":\"www.jupiterone.io\",\"duplicate\":false,\"id\":\"6eac0494-51d2-49be-8da1-ea6649bb50f8\",\"primary\":false,\"redirect_to\":{\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"name\":\"jupiterone.com\"}},{\"name\":\"jupiterone.io\",\"duplicate\":false,\"id\":\"b83dbe62-1698-4834-99ea-3bf38e12866b\",\"primary\":false,\"redirect_to\":{\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"name\":\"jupiterone.com\"}},{\"name\":\"www.jupiterone.com\",\"duplicate\":false,\"id\":\"0de8bdf6-2637-431c-9a2e-eb66a1f107e4\",\"primary\":false,\"redirect_to\":{}},{\"name\":\"jupiterone.com\",\"duplicate\":false,\"id\":\"c2663ce5-e1bf-42d3-8b78-89b032f1dd03\",\"primary\":true,\"redirect_to\":{}},{\"name\":\"jupiterone.wpengine.com\",\"duplicate\":false,\"id\":\"d1806903-21cd-4493-9dba-17968f347ffe\",\"primary\":false,\"redirect_to\":{}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "content-length",
              "value": "804"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "date",
              "value": "Mon, 05 Jul 2021 11:52:24 GMT"
            },
            {
              "name": "x-amzn-requestid",
              "value": "88ee4278-7126-49f8-88d1-85ec7751a475"
            },
            {
              "name": "access-control-allow-origin",
              "value": "*"
            },
            {
              "name": "access-control-allow-headers",
              "value": "Content-Type,X-Amz-Date,Authorization,X-Api-Key"
            },
            {
              "name": "x-amz-apigw-id",
              "value": "B_rSzH9SIAMFbew="
            },
            {
              "name": "access-control-allow-methods",
              "value": "*"
            },
            {
              "name": "x-amzn-trace-id",
              "value": "Root=1-60e2f278-2fc13f37529911bf37e226f7"
            },
            {
              "name": "x-cache",
              "value": "Miss from cloudfront"
            },
            {
              "name": "via",
              "value": "1.1 183f775990c096ea769d4097f23cbad3.cloudfront.net (CloudFront)"
            },
            {
              "name": "x-amz-cf-pop",
              "value": "MNL50-C1"
            },
            {
              "name": "x-amz-cf-id",
              "value": "6NVbpdK8Dz0zQXn_TXZDKRfMavqa3PueOcjb9XEujPSFL5A7E1-GSA=="
            }
          ],
          "headersSize": 601,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-07-05T11:52:23.662Z",
        "time": 1064,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 1064
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
