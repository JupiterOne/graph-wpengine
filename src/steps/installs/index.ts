import {
  createDirectRelationship,
  IntegrationStep,
  IntegrationStepExecutionContext,
  RelationshipClass,
} from '@jupiterone/integration-sdk-core';

import { IntegrationConfig } from '../../config';
import { createAPIClient } from '../../client';
import { Entities, Relationships, IntegrationSteps } from '../constants';
import { getAccountKey } from '../accounts/converters';
import { getSiteKey } from '../sites/converters';
import { createInstallEntity } from './converters';

export async function fetchInstalls({
  instance,
  jobState,
}: IntegrationStepExecutionContext<IntegrationConfig>) {
  const apiClient = createAPIClient(instance.config);

  await apiClient.iterateInstalls(async (install) => {
    const installEntity = createInstallEntity(install);

    await jobState.addEntity(installEntity);

    const accountEntity = await jobState.findEntity(
      getAccountKey(install.account.id),
    );

    const siteEntity = await jobState.findEntity(getSiteKey(install.site.id));

    if (installEntity && accountEntity) {
      await jobState.addRelationship(
        createDirectRelationship({
          _class: RelationshipClass.HAS,
          from: accountEntity,
          to: installEntity,
        }),
      );
    }

    if (installEntity && siteEntity) {
      await jobState.addRelationship(
        createDirectRelationship({
          _class: RelationshipClass.HAS,
          from: siteEntity,
          to: installEntity,
        }),
      );
    }
  });
}

export const installSteps: IntegrationStep<IntegrationConfig>[] = [
  {
    id: IntegrationSteps.INSTALLS,
    name: 'Fetch Installs',
    entities: [Entities.INSTALL],
    relationships: [
      Relationships.ACCOUNT_HAS_INSTALL,
      Relationships.SITE_HAS_INSTALL,
    ],
    dependsOn: [IntegrationSteps.ACCOUNTS, IntegrationSteps.SITES],
    executionHandler: fetchInstalls,
  },
];
