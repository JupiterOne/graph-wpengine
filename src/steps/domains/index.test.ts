import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';
import { IntegrationConfig } from '../../config';
import { fetchDomains } from '.';
import { integrationConfig } from '../../../test/config';
import { setupWPEngineRecording } from '../../../test/recording';
import { Relationships } from '../constants';
import { fetchInstalls } from '../installs';

describe('#fetchDomains', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupWPEngineRecording({
      directory: __dirname,
      name: 'fetchDomains',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchInstalls(context);
    await fetchDomains(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const domains = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Domain'),
    );
    expect(domains.length).toBeGreaterThan(0);
    expect(domains).toMatchGraphObjectSchema({
      _class: ['Domain'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'wp_engine_domain' },
          domainName: { type: 'string' },
          duplicate: { type: 'boolean' },
          primary: { type: 'boolean' },
          redirectTo: { type: 'string' },
        },
      },
    });

    const installs = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Application'),
    );
    expect(installs.length).toBeGreaterThan(0);
    expect(installs).toMatchGraphObjectSchema({
      _class: ['Application'],
      schema: {
        additionalProperties: false,
        properties: {
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          _type: { const: 'wp_engine_install' },
          name: { type: 'string' },
          phpVersion: { type: 'string' },
          status: { type: 'boolean' },
          cname: { type: 'string' },
          stableIps: { type: 'array', items: { type: 'string' } },
          environment: { type: 'string' },
          primaryDomain: { type: 'string' },
          isMultisite: { type: 'boolean' },
        },
      },
    });

    expect(
      context.jobState.collectedRelationships.filter(
        (e) => e._type === Relationships.INSTALL_HAS_DOMAIN._type,
      ),
    ).toMatchDirectRelationshipSchema({
      schema: {
        properties: {
          _class: { const: 'HAS' },
          _type: {
            const: 'wp_engine_install_has_domain',
          },
        },
      },
    });
  });
});
