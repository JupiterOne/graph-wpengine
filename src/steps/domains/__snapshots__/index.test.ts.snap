// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchDomains should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Application",
      ],
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "account": Object {
              "id": "77815c5d-e25c-41a3-ba2d-29f8ceb82ac1",
            },
            "cname": "jupiterone.wpengine.com",
            "environment": "production",
            "id": "30c47931-a91d-468a-a5dc-b9711f06c60b",
            "is_multisite": false,
            "name": "jupiterone",
            "php_version": "7.3",
            "primary_domain": "jupiterone.com",
            "site": Object {
              "id": "4d91f2dd-9743-421e-9bcf-20dcf0a195b2",
            },
            "stable_ips": null,
            "status": "active",
          },
        },
      ],
      "_type": "wp_engine_install",
      "active": true,
      "cname": "jupiterone.wpengine.com",
      "createdOn": undefined,
      "displayName": "jupiterone",
      "environment": "production",
      "id": "30c47931-a91d-468a-a5dc-b9711f06c60b",
      "isMultisite": false,
      "name": "jupiterone",
      "phpVersion": "7.3",
      "primaryDomain": "jupiterone.com",
      "stableIps": Array [],
      "status": true,
    },
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "wp_engine_domain:6eac0494-51d2-49be-8da1-ea6649bb50f8",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "duplicate": false,
            "id": "6eac0494-51d2-49be-8da1-ea6649bb50f8",
            "name": "www.jupiterone.io",
            "primary": false,
            "redirect_to": Object {
              "id": "c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
              "name": "jupiterone.com",
            },
          },
        },
      ],
      "_type": "wp_engine_domain",
      "createdOn": undefined,
      "displayName": "www.jupiterone.io",
      "domainName": "www.jupiterone.io",
      "duplicate": false,
      "id": "6eac0494-51d2-49be-8da1-ea6649bb50f8",
      "name": "www.jupiterone.io",
      "primary": false,
      "redirectTo": "jupiterone.com",
    },
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "wp_engine_domain:b83dbe62-1698-4834-99ea-3bf38e12866b",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "duplicate": false,
            "id": "b83dbe62-1698-4834-99ea-3bf38e12866b",
            "name": "jupiterone.io",
            "primary": false,
            "redirect_to": Object {
              "id": "c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
              "name": "jupiterone.com",
            },
          },
        },
      ],
      "_type": "wp_engine_domain",
      "createdOn": undefined,
      "displayName": "jupiterone.io",
      "domainName": "jupiterone.io",
      "duplicate": false,
      "id": "b83dbe62-1698-4834-99ea-3bf38e12866b",
      "name": "jupiterone.io",
      "primary": false,
      "redirectTo": "jupiterone.com",
    },
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "wp_engine_domain:0de8bdf6-2637-431c-9a2e-eb66a1f107e4",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "duplicate": false,
            "id": "0de8bdf6-2637-431c-9a2e-eb66a1f107e4",
            "name": "www.jupiterone.com",
            "primary": false,
            "redirect_to": Object {},
          },
        },
      ],
      "_type": "wp_engine_domain",
      "createdOn": undefined,
      "displayName": "www.jupiterone.com",
      "domainName": "www.jupiterone.com",
      "duplicate": false,
      "id": "0de8bdf6-2637-431c-9a2e-eb66a1f107e4",
      "name": "www.jupiterone.com",
      "primary": false,
      "redirectTo": undefined,
    },
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "wp_engine_domain:c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "duplicate": false,
            "id": "c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
            "name": "jupiterone.com",
            "primary": true,
            "redirect_to": Object {},
          },
        },
      ],
      "_type": "wp_engine_domain",
      "createdOn": undefined,
      "displayName": "jupiterone.com",
      "domainName": "jupiterone.com",
      "duplicate": false,
      "id": "c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
      "name": "jupiterone.com",
      "primary": true,
      "redirectTo": undefined,
    },
    Object {
      "_class": Array [
        "Domain",
      ],
      "_key": "wp_engine_domain:d1806903-21cd-4493-9dba-17968f347ffe",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "duplicate": false,
            "id": "d1806903-21cd-4493-9dba-17968f347ffe",
            "name": "jupiterone.wpengine.com",
            "primary": false,
            "redirect_to": Object {},
          },
        },
      ],
      "_type": "wp_engine_domain",
      "createdOn": undefined,
      "displayName": "jupiterone.wpengine.com",
      "domainName": "jupiterone.wpengine.com",
      "duplicate": false,
      "id": "d1806903-21cd-4493-9dba-17968f347ffe",
      "name": "jupiterone.wpengine.com",
      "primary": false,
      "redirectTo": undefined,
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b|has|wp_engine_domain:6eac0494-51d2-49be-8da1-ea6649bb50f8",
      "_toEntityKey": "wp_engine_domain:6eac0494-51d2-49be-8da1-ea6649bb50f8",
      "_type": "wp_engine_install_has_domain",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b|has|wp_engine_domain:b83dbe62-1698-4834-99ea-3bf38e12866b",
      "_toEntityKey": "wp_engine_domain:b83dbe62-1698-4834-99ea-3bf38e12866b",
      "_type": "wp_engine_install_has_domain",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b|has|wp_engine_domain:0de8bdf6-2637-431c-9a2e-eb66a1f107e4",
      "_toEntityKey": "wp_engine_domain:0de8bdf6-2637-431c-9a2e-eb66a1f107e4",
      "_type": "wp_engine_install_has_domain",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b|has|wp_engine_domain:c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
      "_toEntityKey": "wp_engine_domain:c2663ce5-e1bf-42d3-8b78-89b032f1dd03",
      "_type": "wp_engine_install_has_domain",
      "displayName": "HAS",
    },
    Object {
      "_class": "HAS",
      "_fromEntityKey": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b",
      "_key": "wp_engine_install:30c47931-a91d-468a-a5dc-b9711f06c60b|has|wp_engine_domain:d1806903-21cd-4493-9dba-17968f347ffe",
      "_toEntityKey": "wp_engine_domain:d1806903-21cd-4493-9dba-17968f347ffe",
      "_type": "wp_engine_install_has_domain",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "wp_engine_install",
    "wp_engine_domain",
    "wp_engine_install_has_domain",
  ],
  "numCollectedEntities": 6,
  "numCollectedRelationships": 5,
}
`;
